app/src/main/java/com/android/sample/model/ingredient/SearchIngredientViewModel.kt

```diff
@@ -63,16 +63,24 @@ interface SearchIngredientViewModel {
     if (quantity1 == null || quantity2 == null) {
       return quantity1 ?: quantity2 ?: ""
     }
+    // Regular expression to find the first number in the string, considering possible decimal
+    // separators
+    val regex = Regex("""\d+[,.]?\d*""")
 
-    // Regular expression to find the first integer in the string
-    val regex = Regex("""\d+""")
     val match1 = regex.find(quantity1)
     val match2 = regex.find(quantity2)
 
-    // If both strings contain an integer, add them together
     return if (match1 != null && match2 != null) {
-      val addition = match1.value.toInt() + match2.value.toInt()
-      quantity1.replaceFirst(match1.value, addition.toString())
+      // Replace ',' with '.' for both matches to standardize
+      val number1 = match1.value.replace(',', '.').toDouble()
+      val number2 = match2.value.replace(',', '.').toDouble()
+      val addition = number1 + number2
+
+      // Format the result, replace the first occurrence, delete .0 and ,0 and return
+      quantity1
+          .replaceFirst(match1.value, addition.toString())
+          .replaceFirst(".0", "")
```

Comment:
I’m having a little trouble understanding this one. If I understand correctly, this issue occurs when we add the same ingredient but with different quantities. The variable `val addition` is the sum of the two quantities, which should be a `Double` and therefore include a `.`. For styling purposes, you want to remove the decimal point and fraction if the result is an integer, as this is visually more appealing.

If I’m not mistaken, the addition of two `Double` values will always result in a `Double`, meaning it’s impossible for the result to contain a `,` instead of a `.`. So, I’m wondering why we’re trying to handle this case with `.replaceFirst(",0", "")`. 

Maybe I’m missing something here, so please don’t hesitate to clarify!

Reply:
Yes, you're right. This line is useless. I've remove it !